File: app/src/main/java/com/silentgames/silent_planet/model/entities/space/fractions/RobotShip.java
package com.silentgames.silent_planet.model.entities.space.fractions;

import android.content.res.Resources;

import com.silentgames.silent_planet.R;
import com.silentgames.silent_planet.logic.Fractions;
import com.silentgames.silent_planet.model.entities.space.SpaceShip;
import com.silentgames.silent_planet.utils.BitmapEditor;

/**
 * Created by gidroshvandel on 24.09.16.
 */
public class RobotShip extends SpaceShip {
    public RobotShip(Resources res) {
        super.setBitmap(BitmapEditor.getEntityBitmap(R.drawable.robot_space_ship, res));
        super.setCanFly(true);
        super.setFraction(Fractions.Robots);
    }
}

File: app/src/main/java/com/silentgames/silent_planet/logic/ecs/entity/unit/space/RobotShip.kt
package com.silentgames.silent_planet.logic.ecs.entity.unit.space

import android.content.Context
import com.silentgames.silent_planet.R
import com.silentgames.silent_planet.logic.ecs.component.Description
import com.silentgames.silent_planet.logic.ecs.component.Texture
import com.silentgames.silent_planet.model.Axis
import com.silentgames.silent_planet.model.fractions.FractionsType
import com.silentgames.silent_planet.utils.BitmapEditor

class RobotShip(
        context: Context,
        position: Axis
) : SpaceShip(
        context,
        position,
        Texture(BitmapEditor.getEntityBitmap(context, R.drawable.robot_space_ship)),
        FractionsType.ROBOT,
        Description(
                context.getString(R.string.robot_ship_name),
                context.getString(R.string.robot_ship_description)
        )
)